go/libraries/doltcore/servercfg/yaml_config.go

```diff
@@ -129,6 +129,7 @@ type YAMLConfig struct {
 	EncodeLoggedQuery *bool                  `yaml:"encode_logged_query,omitempty"`
 	BehaviorConfig    BehaviorYAMLConfig     `yaml:"behavior"`
 	UserConfig        UserYAMLConfig         `yaml:"user"`
+	SkipRootUserInit  *bool                  `yaml:"skip_root_user_initialization,omitempty"`
```

Comment:
I don't think this setting makes sense to persist or specify in a yaml config. If they have a config file they probably want auth, and if not it's easy enough to disable.

Reply:
I changed this so that the flag controlling whether root user initialization is skipped is passed directly as a function argument, instead of including it in `ServerConfig`. (See 0ff27bb1ee6288f76fd68675804d33b4eef42353)